### ISSUE TITLE
Optional elements are not taking into account on generate URI

### ISSUE THREAD

**Issue by martinezdelariva** [zendframework/zend-expressive-fastroute#15]
Hello,

It seems that the generation of a URI with optional suffix using FastRoute is not taking into account to remove the trailing `[, ]` when the optional parameter is given, and remove the whole element when is not provided:

How to reproduce in a Expressive application:

``` php
// bootstrap $app using FastRoute as router

$app->get('/users[/{id}]', function($request, $response, $out) use ($container) {
    /** @var Zend\Expressive\Helper\UrlHelper $helper */
    $router = $container->get(Zend\Expressive\Router\RouterInterface::class);

    return new \Zend\Diactoros\Response\JsonResponse([
        'uri'   => $router->generateUri('users'),
        'uriId' => $router->generateUri('users', ['id' => 1234])
    ]);

}, 'users');

// Output
{"uri":"/users[/{id}]","uriId":"/users[/1234]"}

// Output expected
{"uri":"/users","uriId":"/users/1234"}
```

How to reproduce in a naively way:

``` php
$routerCollector = new FastRoute\RouteCollector(
    new FastRoute\RouteParser\Std(),
    new FastRoute\DataGenerator\GroupCountBased()
);
$router = new Zend\Expressive\Router\FastRouteRouter($routerCollector);
$router->addRoute(
    new Zend\Expressive\Router\Route('/users[/{id}]', 
    function ($req, $res) {}, 
    ['GET'], 
    'users')
);

echo $router->generateUri('users');
echo $router->generateUri('users', ['id' => 1234]);

// Output:
// /users[/{id}]
// /users[/1234]

// Output expected:
// /users
// /users/1234
```

Thanks!

**Comment by schnittstabil** [comment]
Same as #12 !?

**Comment by martinezdelariva** [comment]
@schnittstabil Thanks for your quick response. 
I was pointing in composer to `^1.1`.  Pointing to `dev-master` works as it uses latest change.
Thanks!

**Comment by geerteltink** [comment]
This costed me an hour as well this morning until I realized it was already fixed in dev-master a few months ago.

@weierophinney Can you please merge #11 and #14 and create a new release?

**Comment by hdimo** [comment]
it was fixed a time ago but its not yet merged,
the method changed is
https://github.com/weierophinney/zend-expressive-fastroute/blob/master/src/FastRouteRouter.php#L129
use this version

**Comment by geerteltink** [comment]
@jkhaled The fix is merged into dev-master with #4. It's just not tagged as a release yet.

**Comment by hdimo** [comment]
@xtreamwayz  yes, thanks
